code/modules/power/pt_laser.dm

```diff
@@ -440,6 +453,10 @@
 
 
 /obj/machinery/power/pt_laser/proc/update_output()
+	if(ON_COOLDOWN(src, LASER_COOLDOWN_ID, src.laser_cooldown))
```

Comment:
The way this is currently written seems to make turning the beam on instant, while turning it off takes time. This seems like the opposite of what we'd want.

Reply:
If it's off cooldown, this will make any action, turning off, turning on, or adjusting the power instant. If they attempt to do that when it's on_cooldown it should instead early return and set a flag that makes the next `process` update the laser.

Reply:
An alternative, which I don't mind doing, is removing the `ON_COOLDOWN`, always just setting the `laser_output_needs_update` flag, and always waiting for `process`. We'd lose the snappiness of the current PTL but it would make things a little simpler.

Reply:
Nono the cooldown is fine. Looking into it a bit further it looks like the issue is actually just that `update_output` isn't checking the `online` var to determine whether to disable the beam, so it's always waiting for the next process tick instead.